File: Talleres-2B/Taller-4/Ejercicio_Mancilla3.py
print("\t---Chequeo de disponibilidad de anaqueles de góndolas---")

rd = 0

Disp = []

k = int(input("Ingrese el número de góndolas: "))
n = int(input("Ingrese el número de anaqueles: "))

for i in range (0, k):
    Disp.append([])
    for j in range (0, n):
        valor = int(input("Disp[",str(i),"][",str(j),"]: "))
        Disp[i].append(valor)

# Se hace el chequeo de la disponibilidad
# Del anaquel q en la góndola g

q = int(input("Ingrese el anaquel (q) a evaluar: "))
g = int(input("Ingrese la góndola (g) que necesita comprobar: "))

# Crear la función y usarla en el proces

def FunDisp (Disp, g, q):
    
    if (Disp[g][q] == 0):
        return 0

    else:
        return 1        

rd = FunDisp(Disp, g, q)

if (rd == 0):
    print("El anaquel ",q," en la góndola ",g," está disponible")
else:
    print("El anaquel ",q," en la góndola ",g," NO está disponible")

File: Talleres-2B/Taller-4/FuncionesYProced/src/funcionesyproced/Ejercicio_Mancilla3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package funcionesyproced;
import java.util.Scanner;
/**
 *
 * @author Santiago Espinoza
 */
public class Ejercicio_Mancilla3 {
    public static void main(String[] args) {
        Scanner leer = new Scanner(System.in);
        int k, n, i, j, q, g, rd;
        int Disp[][] = new int [50][50];
        System.out.print("Ingrese el valor de k: ");
        k = leer.nextInt();
        System.out.print("Ingrese el valor de n: ");
        n = leer.nextInt();
        
        for (i = 0; i < k; i++) {
            for (j = 0; j < n; j++) {
                System.out.print("Ingrese el valor de Disp["+i+"]["+j+"]= ");
                Disp[i][j]=leer.nextInt();
                
            }
        }
        
        // Se hace el chequeo de la disponibilidad
        // del anaquel q en la gondola g
        System.out.print("Ingrese el valor de q: ");
        q = leer.nextInt();
        System.out.print("Ingrese el valor de g: ");
        g = leer.nextInt();
        rd = FunDisp(Disp,g,q);
        if(rd==0){
            System.out.println("El anaquel "+q+" en la gondola "+g+" está disponible");
        }
        else{
            System.out.println("El anaquel "+q+" en la gondola "+g+" no está disponible");
        }
        
    }
    public static int FunDisp(int [][]Disp,int g, int q){
        if(Disp[g][q]==0){
            return 0;
        }
        else{
            return 1;
        }
    }
}
